protostar/cairo_testing/execution_environments/cairo_test_execution_environment.py

```diff
@@ -59,3 +72,10 @@ def _get_hint_locals(self, state: CairoTestExecutionState) -> HintLocalsDict:
             hint_locals[custom_hint_local.name] = custom_hint_local.build()
 
         return hint_locals
+
+    def _wrap_hint_exceptions(self, ex: Exception):
+        if isinstance(ex, ExpectEventsMismatchException):
+            return ExpectEventsMismatchReportedException(
+                message=ex.message, event_matching_result=ex.matching_result
+            )
+        return ex
```

Comment:
Can you explain the purpose of this function? It looks totally redundant. What you do is you catch `ExpectEventsMismatchException` and then you conditionally re-raise either `ExpectEventsMismatchException` or `ExpectEventsMismatchReportedException` if the original error is `ExpectEventsMismatchException`, which happens in 100% situations.

Reply:
Using `ExpectEventsMismatchReportedException` in the controller would create a circular dependency.

Reply:
why not just "ReportedException" ? We don't seem to catch it anywhere anyways

Reply:
Ok, I'll change that. I thought `ReportedException` is in the `protostar.testing` (which actually would make more sense) not in the `protostar.starknet`. Circular dependency won't be created.

Reply:
That doesn't change the fact that you can simply do
```
except ExpectEventsMismatchException as ex:
                raise ExpectEventsMismatchReportedException(...)
```
in `execute_test_case` and remove `_wrap_hint_exceptions`

Reply:
[use one ExpectEventsMismatch exception](https://github.com/software-mansion/protostar/pull/1412/commits/d2f9490978db998738a2896fc8ba6232831a47b3)